File: sorting-examples/SelectionSort.java
public class SelectionSort
{
    public static void main(String[] args)
    {
        int[] arr = {64, 25, 12, 22, 11};
        descendingSort(arr);
        for (int i = 0; i < 5; i++)
        System.out.print(arr[i] + " ");
    }
    
    private static void descendingSort(int arr[])
    {
        int max_idx;
        for (int i = 0; i < arr.length - 1; i++)
        {
            max_idx = i;
            for (int j = i + 1; j < arr.length; j++)
            {
                if (arr[j] > arr[max_idx])
                    max_idx = j;
            }
            int temp = arr[max_idx];
            arr[max_idx] = arr[i];
            arr[i] = temp;
        }
    }

    private static void ascendingSort(int arr[])
    {
        int max_idx;
        for (int i = 0; i < arr.length - 1; i++)
        {
            max_idx = i;
            for (int j = i + 1; j < arr.length; j++)
            {
                if (arr[j] < arr[max_idx])
                    max_idx = j;
            }
            int temp = arr[max_idx];
            arr[max_idx] = arr[i];
            arr[i] = temp;
        }
    }
}

File: sorting-examples/selection_sort.cpp
// David Kanter Eivin

#include<iostream>

void swap(int *a, int *b)
{
    int temp = *a;
    *a = *b;
    *b = temp;
}

void ascendingSort(int arr[], int len)
{
    int max_idx;
    for (int i = 0; i < len - 1; i++)
    {
        max_idx = i;
        for (int j = i + 1; j < len; j++)
        {
            if (arr[j] < arr[max_idx])
                max_idx = j;
        }
        swap(&arr[max_idx], &arr[i]);
    }
}

void descendingSort(int arr[], int len)
{
    int max_idx;
    for (int i = 0; i < len - 1; i++)
    {
        max_idx = i;
        for (int j = i + 1; j < len; j++)
        {
            if (arr[j] > arr[max_idx])
                max_idx = j;
        }
        swap(&arr[max_idx], &arr[i]);
    }
}

int main() {
    int arr[5] = {64,25,12,22,11};
    descendingSort(arr, 5);
    for (int i = 0; i < 5; i++)
        std::cout << arr[i] << " ";
    return 0;
}